src/Providers/SoapServiceProvider.php

```diff
@@ -1,37 +1,35 @@
 <?php
 
+declare(strict_types=1);
+
 namespace RicorocksDigitalAgency\Soap\Providers;
 
+use Illuminate\Contracts\Foundation\Application;
 use Illuminate\Support\ServiceProvider;
-use RicorocksDigitalAgency\Soap\Parameters\Builder;
+use RicorocksDigitalAgency\Soap\Contracts\Builder;
+use RicorocksDigitalAgency\Soap\Contracts\Request;
 use RicorocksDigitalAgency\Soap\Parameters\IntelligentBuilder;
-use RicorocksDigitalAgency\Soap\Ray\SoapWatcher;
-use RicorocksDigitalAgency\Soap\Request\Request;
-use RicorocksDigitalAgency\Soap\Request\SoapClientRequest;
+use RicorocksDigitalAgency\Soap\Request\SoapPhpRequest;
 use RicorocksDigitalAgency\Soap\Soap;
+use RicorocksDigitalAgency\Soap\Support\SoapClients\DecoratedClient;
 
-class SoapServiceProvider extends ServiceProvider
+/**
+ * @internal
+ */
+final class SoapServiceProvider extends ServiceProvider
 {
-    public function register()
+    public function register(): void
     {
-        $this->app->singleton('soap', fn () => app(Soap::class));
-        $this->app->bind(Request::class, SoapClientRequest::class);
+        $this->app->singleton('soap', fn (Application $app) => $app->make(Soap::class));
         $this->app->bind(Builder::class, IntelligentBuilder::class);
-
-        $this->registerRay();
+        $this->app->bind(Request::class, fn (Application $app) => new SoapPhpRequest(
+            $app->make(Builder::class),
+            new DecoratedClient()
```

Comment:
So if I understand this correctly, changing a client would mean overwriting the bind to the request class?
The request class itself should be compatible over different clients?